spirv_cpp.cpp

```diff
@@ -36,6 +36,64 @@ void CompilerCPP::emit_buffer_block(const SPIRVariable &var)
 	statement("");
 }
 
+string CompilerCPP::type_to_array_cpp(const SPIRType &type)
+{
+	string res;
+	for (auto &size : type.array)
+	{
+		res += "[";
+		if (size)
+			res += convert_to_string(size);
+		else
+			res += "1"; // Flexible array member is prohibited in C++11.
+		res += "]";
+	}
+	return res;
+}
+
+string CompilerCPP::member_decl(const SPIRType &type, const SPIRType &membertype, uint32_t index)
+{
+	uint64_t memberflags = 0;
+	auto &memb = meta[type.self].members;
+	if (index < memb.size())
+		memberflags = memb[index].decoration_flags;
+
+	return join(layout_for_member(type, index), flags_to_precision_qualifiers_glsl(membertype, memberflags),
+	            type_to_glsl(membertype), " ", to_member_name(type, index), type_to_array_cpp(membertype));
+}
+
+void CompilerCPP::emit_struct(const SPIRType &type)
```

Comment:
Is there anything C++ specific about this method? It's not marked as override or virtual, so it looks like it can be removed?